src/main/java/edu/ucsb/cs156/happiercows/controllers/CowDeathController.java

```diff
@@ -0,0 +1,125 @@
+package edu.ucsb.cs156.happiercows.controllers;
+
+import java.time.LocalDateTime;
+import java.util.ArrayList;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.ObjectMapper;
+
+import io.swagger.annotations.Api;
+import io.swagger.annotations.ApiOperation;
+import io.swagger.annotations.ApiParam;
+
+import lombok.extern.slf4j.Slf4j;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.format.annotation.DateTimeFormat;
+import org.springframework.http.ResponseEntity;
+import org.springframework.security.access.prepost.PreAuthorize;
+
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RequestParam;
+import org.springframework.web.bind.annotation.RestController;
+
+import edu.ucsb.cs156.happiercows.errors.EntityNotFoundException;
+
+import edu.ucsb.cs156.happiercows.entities.Commons;
+import edu.ucsb.cs156.happiercows.entities.UserCommons;
+import edu.ucsb.cs156.happiercows.entities.CowDeath;
+import edu.ucsb.cs156.happiercows.repositories.CommonsRepository;
+import edu.ucsb.cs156.happiercows.repositories.UserCommonsRepository;
+import edu.ucsb.cs156.happiercows.repositories.CowDeathRepository;
+
+
+@Slf4j
+@Api(description = "Cow Death")
+@RequestMapping("/api/cowdeath")
+@RestController
+public class CowDeathController extends ApiController {
+    
+    @Autowired
+    private CowDeathRepository cowDeathRepository;
+
+    @Autowired
+    private CommonsRepository commonsRepository;
+
+    @Autowired
+    private UserCommonsRepository userCommonsRepository;
+
+    @Autowired
+    ObjectMapper mapper;
+
+    @ApiOperation(value = "Create a new CowDeath entity")
+    @PreAuthorize("hasRole('ROLE_ADMIN')")
+    @PostMapping(value = "", produces = "application/json")
+    public ResponseEntity<String> createCowDeath(
+        @ApiParam("commons_id") @RequestParam long commonsId,
+        @ApiParam("user_id") @RequestParam long userId,
+        @ApiParam("zonedDateTime") @RequestParam("zonedDateTime") @DateTimeFormat(iso = DateTimeFormat.ISO.DATE_TIME) LocalDateTime zonedDateTime,
```

Comment:
Still need to remove this parameter from the controller.  It is meant to be assigned automatically by the system, not by the end user.

Reply:
@pconrad  I removed the datetime parameter from the controller. I tried setting the date time in the controller using LocalDateTime.now(), as shown in the screenshot below:

![image](https://user-images.githubusercontent.com/39968382/205745902-bb520985-3db0-4631-880a-8d2703d69fa8.png)

But I'm not sure how to fix the tests, since the current time when starting the test is different from the current time when the test runs the controller:

![image](https://user-images.githubusercontent.com/39968382/205746664-f5a15657-69b5-49e9-8fa1-d58d1d10b821.png)

Maybe I could try something with exception handling and check if the two times are close enough (within seconds of each other)?

Also, the parameter zonedDateTime is actually a LocalDateTime object, not a ZonedDateTime object. Not sure if that's an issue. That's how it is in the main branch.